src/offset_vector.c

```diff
@@ -40,26 +42,32 @@ uint32_t _ovi_Next(void *ctx, RSQueryTerm **t);
 void _ovi_Rewind(void *ctx);
 
 /* memory pool for buffer iterators */
-static mempool_t *__offsetIters = NULL;
-static mempool_t *__aggregateIters = NULL;
+static pthread_key_t __offsetIters;
+static pthread_key_t __aggregateIters;
+
+static void __attribute__((constructor)) initKeys() {
```

Comment:
Where is this used? what does `__attribute__((constructor))` means?

Reply:
copied from a similar function in `values.c`. it means that this function is called when the library is loaded (I suppose right after the globals are set)